sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -352,13 +352,31 @@ static int _fsm_call_close(gnrc_tcp_tcb_t *tcb)
 /**
  * @brief FSM handling function for forcefull connection teardown sequence.
  *
- * @returns   -EOPNOTSUPP. Currently not implemented.
+ * @param[in,out] tcb   TCB holding the connection information.
+ *
+ * @returns   Zero on success.
  */
-static int _fsm_call_abort(void)
+static int _fsm_call_abort(gnrc_tcp_tcb_t *tcb)
 {
+    gnrc_pktsnip_t *out_pkt = NULL;     /* Outgoing packet */
+    uint16_t seq_con = 0;               /* Sequence number consumption of outgoing packet */
+
     DEBUG("gnrc_tcp_fsm.c : _fsm_call_abort()\n");
-    DEBUG("gnrc_tcp_fsm.c : _fsm_call_abort() : ABORT not implemented\n");
-    return -EOPNOTSUPP;
+
+    /* A reset must be sent in case the TCB state is in one of those cases */
+    if (tcb->state == FSM_STATE_SYN_RCVD || tcb->state == FSM_STATE_ESTABLISHED ||
```

Comment:
this might look nicer:

```
switch (tcb->state) {
case FSM_STATE_SYN_RECV:
case FSM_STATE_ESTABLISHED:
...
case FSM_STATE_CLOSE_WAIT:
    gnrc_pktsnip_t *out_pkt = NULL; 
    uint16_t seq_con = 0;   
    /* Send RST packet without retransmit */
    _pkt_build(tcb, &out_pkt, &seq_con, MSK_RST, tcb->snd_nxt, tcb->rcv_nxt, NULL, 0);
    _pkt_send(tcb, out_pkt, seq_con, false);
    break;
default:
    break;
}
```